internal/pkg/cli/env_upgrade.go

```diff
@@ -105,14 +105,14 @@ func newEnvUpgradeOpts(vars envUpgradeVars) (*envUpgradeOpts, error) {
 		newTemplateUpgrader: func(conf *config.Environment) (envTemplateUpgrader, error) {
 			sess, err := sessProvider.FromRole(conf.ManagerRoleARN, conf.Region)
 			if err != nil {
-				return nil, fmt.Errorf("create template session from role %s and region %s: %v", conf.ManagerRoleARN, conf.Region, err)
+				return nil, fmt.Errorf("create session from role %s and region %s: %v", conf.ManagerRoleARN, conf.Region, err)
 			}
 			return cloudformation.New(sess), nil
 		},
-		newS3: func(conf *config.Environment) (uploader, error) {
-			sess, err := sessProvider.FromRole(conf.ManagerRoleARN, conf.Region)
+		newS3: func(region string) (uploader, error) {
+			sess, err := sessProvider.DefaultWithRegion(region)
```

Comment:
we want to keep this part ( the changes in this file), just the `s3` package should be reverted

Reply:
Sorry - updated!